app/models/main.py

```diff
@@ -1653,13 +1656,37 @@ def validate(self, enforce_required=True, required_fields=None, max_day_rate=Non
         if errs:
             raise ValidationError(errs)
 
+    class query_class(BaseQuery):
+        def has_datetime_field_between(self, attr, start_datetime, end_datetime, inclusive=None):
+            """
+            Date filter values should be datetime objects. e.g.
+                BriefResponse.query.has_date_field_between(
+                    'published_at', datetime(2017, 1, 1), datetime(2017, 1, 2, 23, 59, 59, 999999), inclusive=True
+                )
+            would return brief responses published between 2017-01-01T00:00:00.000000Z and 2017-01-02T23:59:59.999999Z.
+            """
+            if not (isinstance(start_datetime, datetime) and isinstance(end_datetime, datetime)):
+                raise ValueError('Datetime object required')
```

Comment:
We often try not to do special type-testing in python - and in this particular case I think it's a test that a) would be the responsibility of SQLAlchemy and b) I suspect SQLAlchemy does it anyway.

Reply:
and c) it would perhaps unnecessarily prevent us from being able to compare `date`s with `datetime` fields.

Reply:
I'll have a look at this - if we can avoid it here we should take it out of the equivalent `Brief.query.has_datetime_field_between` too.

Reply:
If you're looking anyway then it really might be worth considering whether having this kind of function adds much value ;-)

Reply:
Good news everyone: the built-in `Foo.query.mydatefield.between(start_date, end_date)` works just fine if we only want inclusive ranges, which we do here 👍